playbooks/v2_deploy.yml

```diff
@@ -0,0 +1,87 @@
+---
+
+- name: v2 deploy
+  hosts: ofn_servers
+  remote_user: "{{ user }}"
+  become: yes
+  become_user: "{{ unicorn_user }}"
+
+  roles:
+    # Include and enable only
+    - role: maintenance_mode
+      become: yes
+      become_user: root
+
+  handlers:
+    - include: ../roles/shared_handlers/handlers/main.yml
+
+  tasks:
+    - block:
+      - name: stop unicorn
+        service:
+          name: unicorn_{{ app }}
+          state: stopped
+        become: yes
+        become_user: root
+
+      - name: stop delayed job
+        service:
+          name: delayed_job_{{ app }}
+          state: stopped
+        become: yes
+        become_user: root
+
+      - include_role:
+          name: deploy
+
+      - meta: flush_handlers # Ensure handlers run successfully before reporting success
+      - name: refresh products cache
+        command: "{{ bundle_path }} exec rake ofn:cache:warm_products"
+        args:
+          chdir: "{{ current_path }}"
+
+      # Include and disable only
+      - name: disable maintenance mode
+        file:
+          dest: "/etc/nginx/maintenance.html"
+          state: absent
+        become: yes
+        become_user: root
+
+      - name: Notify Slack of successful deployment
```

Comment:
@Matt-Yorkley should we turn these into a role?

Reply:
Yeah, it might be nice to extract this stuff now. It's still quite specific to the deploy role, so maybe we could just put these tasks in `roles/deploy/tasks/success.yml` (and failure.yml) and use it where needed with: 
```
-  name: success messages
   include_role:
     name: deploy
     tasks_from: success.yml
```
What do you think?

Reply:
Or we could do something similar, but arranged like: `roles/notifications/tasks/deploy_success.yml`,  `roles/notifications/tasks/deploy_failure.yml`,  `roles/notifications/tasks/provision_success.yml` ...?

Reply:
Considering this will die as soon as we deploy v2 to AU it might be better to wait until next time we end up duplicating it, agree?